activerecord/lib/active_record/connection_adapters/mysql/database_statements.rb

```diff
@@ -45,7 +45,11 @@ def execute(sql, name = nil, async: false)
           # made since we established the connection
           @connection.query_options[:database_timezone] = ActiveRecord::Base.default_timezone
 
-          super
+          if add_query_comment?
+            sql = add_comment_to_sql(sql)
+          end
```

Comment:
Given this is the same pattern in all instances, maybe the whole block should be encapsulated in a single method call?